core/object/worker_thread_pool.cpp

```diff
@@ -788,6 +827,9 @@ void WorkerThreadPool::init(int p_thread_count, float p_low_priority_task_ratio)
 
 	print_verbose(vformat("WorkerThreadPool: %d threads, %d max low-priority.", p_thread_count, max_low_priority_threads));
 
+	// Reserve 5 threads in case we need separate threads for 1) 2D physics 2) 3D physics 3) rendering 4) GPU texture compression, 5) all other tasks.
+	// We cannot safely increase the Vector size at runtime, so reserve enough up front, but only launch those needed.
+	threads.reserve(5);
```

Comment:
@bruvzg @Ivorforce This is my hack to ensure that the LocalVector is never resized at run time. Personally, I think this is a bit brittle, but I'm not sure what else to do. Reading from the project settings like I suggested above seems equally brittle to me. 

One option that may be a bit safer would be to do this change, but use a FixedVector instead of a LocalVector, then if we ever try to allocate more threads than those reserved, we will get a very clear error and it will be easy to trace back the issue to the number of threads reserved. 

What do you think?

Reply:
I don't think `FixedVector` can work, because `threads` needs to have a dynamic size according to `WorkerThreadPool::init`.

The regular 'safe' resize option would be to change `threads` to hold `ThreadData *` instead of `ThreadData`. On indexed access, this would mean an additional dereference. I don't think it will lead to noticeable regressions, but I can't' guarantee it.

The `reserve` trick also seems OK for me. It would hold until we get more pump threads, which kinda seems like a ticking time bomb, but I'm also not sure we'll have this system forever anyway. I don't fully understand pump tasks yet but my impression so far has been sceptical.